js/cryptoLib/index.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+const electron = require('electron');
+const app = electron.app;
+const path = require('path');
+const fs = require('fs');
+const archiver = require('archiver');
+const zipArchive = archiver('zip');
+const extract = require('extract-zip');
+const isDevEnv = require('../utils/misc.js').isDevEnv;
+const crypto = require('./crypto');
+
+const userData = path.join(app.getPath('userData'));
+const INDEX_DATA_FOLDER = isDevEnv ? './data/search_index' : path.join(userData, 'data/search_index');
+const TEMPORARY_PATH = isDevEnv ? path.join(__dirname, '..', '..') : userData;
+
+class Crypto {
+
+    // TODO: Need to pass key for encryption and decryption
+    constructor() {
+
+        // will be handling after implementing in client app
+        let userId = 'user_data';
+        let INDEX_VERSION = 'v1';
+        // will be handling after implementing in client app
+
+        this.indexDataFolder = INDEX_DATA_FOLDER + '_' + userId + '_' + INDEX_VERSION;
+        this.dump = TEMPORARY_PATH;
```

Comment:
`userData = path.join(app.getPath('userData'));`
`TEMPORARY_PATH = isDevEnv ? path.join(__dirname, '..', '..') : userData;`

Yes, In production environment it will be user's config path.